src/modules/luksopenswaphookcfg/Tests.cpp

```diff
@@ -0,0 +1,247 @@
+/* === This file is part of Calamares - <https://calamares.io> ===
+ *
+ *  SPDX-FileCopyrightText: 2021 Adriaan de Groot <groot@kde.org>
+ *  SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ */
+
+#include "LOSHInfo.h"
+#include "LOSHJob.h"
+
+#include "GlobalStorage.h"
+#include "JobQueue.h"
+#include "utils/CalamaresUtilsSystem.h"
+#include "utils/Logger.h"
+
+#include <QtTest/QtTest>
+
+// LOSH = LUKS Open Swap Hook (Job)
+
+// Implementation details
+extern QString get_assignment_part( const QString& line );
+extern void write_openswap_conf( const QString& path, QStringList& contents, const LOSHInfo& info );
+
+class LOSHTests : public QObject
+{
+    Q_OBJECT
+public:
+    LOSHTests();
+    ~LOSHTests() override {}
+
+private Q_SLOTS:
+    void initTestCase();
+
+    void testAssignmentExtraction_data();
+    void testAssignmentExtraction();
+
+    void testLOSHInfo();
+    void testConfigWriting();
+    void testJob();
+};
+
+LOSHTests::LOSHTests() {}
+
+void
+LOSHTests::initTestCase()
+{
+    Logger::setupLogLevel( Logger::LOGDEBUG );
+    cDebug() << "LOSH test started.";
+}
+
+
+void
+LOSHTests::testAssignmentExtraction_data()
+{
+    QTest::addColumn< QString >( "line" );
+    QTest::addColumn< QString >( "match" );
+
+    QTest::newRow( "empty" ) << QString() << QString();
+    QTest::newRow( "comment-only1" ) << QStringLiteral( "#  " ) << QString();
+    QTest::newRow( "comment-only2" ) << QStringLiteral( "###" ) << QString();
+    QTest::newRow( "comment-only3" ) << QStringLiteral( "# # #" ) << QString();
+
+    QTest::newRow( "comment-text" ) << QStringLiteral( "#   NOTE:" ) << QString();
+    QTest::newRow( "comment-story" ) << QStringLiteral( "# This is a shell comment" ) << QString();
+    // We look for assignments, but only for single-words
+    QTest::newRow( "comment-space-eq" ) << QStringLiteral( "# Check that a = b" ) << QString();
+
+
+    QTest::newRow( "assignment1" ) << QStringLiteral( "a=1" ) << QStringLiteral( "a" );
+    QTest::newRow( "assignment2" ) << QStringLiteral( "a = 1" ) << QStringLiteral( "a" );
+    QTest::newRow( "assignment3" ) << QStringLiteral( "#   a=1" ) << QStringLiteral( "a" );
+    QTest::newRow( "assignment4" ) << QStringLiteral( "cows   = 12" ) << QStringLiteral( "cows" );
+    QTest::newRow( "assignment5" ) << QStringLiteral( "#   # cows=1" ) << QStringLiteral( "cows" );
+    QTest::newRow( "assignment6" ) << QStringLiteral( "# moose='cool'  # not cows" ) << QStringLiteral( "moose" );
+    QTest::newRow( "assignment7" ) << QStringLiteral( " moose=cows=42" ) << QStringLiteral( "moose" );
```

Comment:
The default file uses this form for a commented assignment `#a=blah`, it might be a good to have a test case for that.

Reply:
I've added tests for that form as well.